openwisp_firmware_upgrader/tests/test_private_storage.py

```diff
@@ -74,3 +80,47 @@ def test_superuser(self):
         user = self._get_admin()
         self.client.force_login(user)
         self._download_firmware_assert_status(200)
+
+    def test_view_permission_check(self):
+        staff_user = self._get_operator()
+        self.client.force_login(staff_user)
+        org = self.image.build.category.organization
+
+        with self.subTest("Test initial access without permissions"):
+            self._download_firmware_assert_status(403)
```

Comment:
This test is incorrect (or, the description is misleading). 

The `Operator` group has read-only permission for the FirmwareImage model 

https://github.com/openwisp/openwisp-firmware-upgrader/blob/e47bdcf6edb7d2d5db7a0ce0c7198b3dceb60f03/openwisp_firmware_upgrader/migrations/__init__.py#L28-L34

So, this code only tests that the user is not an org admin.